group-coordinator/src/main/java/org/apache/kafka/coordinator/group/generic/GenericGroupMember.java

```diff
@@ -0,0 +1,499 @@
+    }
+
+    /**
+     * @return the member's assignment.
+     */
+    public byte[] assignment() {
+        return assignment;
+    }
+
+    /**
+     * @return the awaiting join callback.
+     */
+    public CompletableFuture<JoinGroupResponseData> awaitingJoinCallback() {
+        return awaitingJoinCallback;
+    }
+
+    /**
+     * @param value the updated join callback.
+     */
+    public void setAwaitingJoinCallback(CompletableFuture<JoinGroupResponseData> value) {
+        this.awaitingJoinCallback = value;
+    }
+
+    /**
+     * @return the awaiting sync callback.
+     */
+    public CompletableFuture<SyncGroupResponseData> awaitingSyncCallback() {
+        return awaitingSyncCallback;
+    }
+
+    /**
+     * @param value the updated sync callback.
+     */
+    public void setAwaitingSyncCallback(CompletableFuture<SyncGroupResponseData> value) {
+        this.awaitingSyncCallback = value;
+    }
+
+    /**
+     * @return true if the member is new, false otherwise.
+     */
+    public boolean isNew() {
+        return isNew;
+    }
+
+    /**
+     * @param value true if the member is new, false otherwise.
+     */
+    public void setIsNew(boolean value) {
+        this.isNew = value;
+    }
+
+    /**
+     * @return true if the existing heartbeat was satisfied, false otherwise.
+     */
+    public boolean heartBeatSatisfied() {
+        return heartbeatSatisfied;
+    }
+
+    /**
+     * @param value whether the heartbeat was satisfied.
+     */
+    public void setHeartBeatSatisfied(boolean value) {
+        this.heartbeatSatisfied = value;
+    }
+
+    @Override
+    public String toString() {
+        return "GenericGroupMember(" +
+            "memberId='" + memberId + '\'' +
+            ", groupInstanceId='" + groupInstanceId + '\'' +
+            ", clientId='" + clientId + '\'' +
+            ", clientHost='" + clientHost + '\'' +
+            ", rebalanceTimeoutMs=" + rebalanceTimeoutMs +
+            ", sessionTimeoutMs=" + sessionTimeoutMs +
+            ", protocolType='" + protocolType + '\'' +
+            ", supportedProtocols=" + supportedProtocols +
+            ')';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        GenericGroupMember that = (GenericGroupMember) o;
+
+        return memberId.equals(that.memberId) &&
+            groupInstanceId.equals(that.groupInstanceId) &&
+            clientId.equals(that.clientId) &&
+            clientHost.equals(that.clientHost) &&
+            rebalanceTimeoutMs == that.rebalanceTimeoutMs &&
+            sessionTimeoutMs == that.sessionTimeoutMs &&
+            protocolType.equals(that.protocolType) &&
+            supportedProtocols.equals(that.supportedProtocols) &&
+            Arrays.equals(assignment, that.assignment);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
```

Comment:
I think that we usually prefer to old style because it does not require to convert primitive types objects. e.g.

```
        final int prime = 31;
        int result = prime + topicId.hashCode();
        result = prime * result + topicPartition.hashCode();
        return result;
```

You can auto generate this with the IDE.

Reply:
Wow, TIL we can generate this :)